### DISCUSSION TITLE
Attempt to read property "users" on null

### DISCUSSION THREAD

**Discussion by eshbanbahadur** [jenssegers/laravel-mongodb#2535] [Q&A]
I am trying to implement 'Many to Many' relationship in a default Laravel project but I only receive the following error:
 "Attempt to read property "users" on null"

I even tried it with custom foreign keys but I did not get desired results and always get error message. I also know that this package not supports pivot tables for Many to Many relationships.

Please guide that how fix this issue

I have two models (user, role) and below is the code of my Models:

// USER MODEL

````````

namespace App\Models;

use Jenssegers\Mongodb\Eloquent\Model;


class user extends Model
{
   // protected $connection = 'mongodb';
    protected $collection = 'users';


    public function roles(){
        return $this->belongsToMany(role::class,null,'role_id_custom');
    }
}

````````

This is for ROLE Model

````````
<?php

namespace App\Models;

#use Illuminate\Database\Eloquent\Factories\HasFactory;
#use Illuminate\Database\Eloquent\Model;

use Jenssegers\Mongodb\Eloquent\Model;

class role extends Model
{

   // protected $connection = 'mongodb';
    protected $collection = 'roles';

    public function users(){
        return $this->belongsToMany(user::class,null);
    }
}
````````

And this is the code of Home Controller:

````````
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;

use App\Models\user;
use App\Models\role;

class home extends Controller
{

    public function index()
    {
        $role = user::find(1)->roles;
        return $role;

       // $user = role::find(1)->users;
        //return $user;
    }
    //
}
````````
Here is the Schema of my MongoDB
````````
USERS
_id:
name:
role_id_custom: (This is a foreign key)

ROLES
_id:
role:
role_id_custom
````````

Please guide where is the issue

Thanks

**Comment by zigzagdev** [comment]
@eshbanbahadur 

Hi! If you haven't solved this problem, try these options !!

Neither of those will work the way you expect, because Eloquent’s update() does not support MongoDB options like multi / multiple, and limit() is not applied to an update() in Eloquent (it turns into a mass update).

If you want to update only one document, do one of these:

- Option 1 (recommended): fetch one model and save

``` php
$product = Product::where(/* your condition */)->first();

if (!$product) {
    return; // or throw
}

$product->xxx = 'yyy';
$product->save();
```

- Option 2: update a single known ID

```php
Product::where('_id', $id)->update(['xxx' => 'yyy']);
```

### Key point:

- Eloquent `update()` is “update all matching docs”, not “update first match only”.
- Mongo-specific options (`multi`, `multiple`) won’t be respected via Eloquent.

If you share what condition you’re using to select the product (and whether _id is ObjectId or integer), I can show the exact best query.
